File: problems/0014-longest-common-prefix/solution.cpp
// solution in 15 minutes
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        
        string pre = "";
        
        for (int i = 0; i < strs[0].size(); i++) {
            char c = strs[0][i];
            
            for (int s = 1; s < strs.size(); s++) {
                if (i >= strs[s].size() || strs[s][i] != c) return pre;
            }
            
            pre.push_back(c);
        }
        return pre;
    }
};

File: problems/0014-longest-common-prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        ans = ""

        for i, c in enumerate(strs[0]):
            for word in strs:
                if i >= len(word) or word[i] != c:
                    return ans
            ans += c
        
        return ans
                

